tests/functional/UseCase/Table/Import/FromTable/ImportTableFromTableTest.php

```diff
@@ -294,6 +298,272 @@ public function testImportTableFromTableFullLoadWithTimestamp(): void
+            $destinationTableName,
+            false,
+            new ColumnCollection([
+                BigqueryColumn::createGenericColumn('col1'),
+                BigqueryColumn::createGenericColumn('col2'),
+            ]),
+            [],
+        );
+        $bqClient->runQuery($bqClient->query($qb->getCreateTableCommand(
+            $destinationDefinition->getSchemaName(),
+            $destinationDefinition->getTableName(),
+            $destinationDefinition->getColumnsDefinitions(),
+            $destinationDefinition->getPrimaryKeysNames(),
+        )));
+
+        $columnMappings = new RepeatedField(
+            GPBType::MESSAGE,
+            TableImportFromTableCommand\SourceTableMapping\ColumnMapping::class,
+        );
+        $columnMappings[] = (new TableImportFromTableCommand\SourceTableMapping\ColumnMapping())
+            ->setSourceColumnName('col1')
+            ->setDestinationColumnName('col1');
+        $columnMappings[] = (new TableImportFromTableCommand\SourceTableMapping\ColumnMapping())
+            ->setSourceColumnName('col2')
+            ->setDestinationColumnName('col2');
+
+        $path = new RepeatedField(GPBType::STRING);
+        $path[] = $bucketDatabaseName;
+        $cmd = (new TableImportFromTableCommand())
+            ->setSource(
+                (new TableImportFromTableCommand\SourceTableMapping())
+                    ->setPath($path)
+                    ->setTableName($sourceTableName)
+                    ->setColumnMappings($columnMappings),
+            )
+            ->setDestination(
+                (new Table())
+                    ->setPath($path)
+                    ->setTableName($destinationTableName),
+            )
+            ->setImportOptions(
+                (new ImportOptions())
+                    ->setImportType(ImportOptions\ImportType::FULL)
+                    ->setDedupType(ImportOptions\DedupType::UPDATE_DUPLICATES)
+                    ->setImportStrategy(ImportOptions\ImportStrategy::USER_DEFINED_TABLE)
+                    ->setConvertEmptyValuesToNullOnColumns(new RepeatedField(GPBType::STRING))
+                    ->setNumberOfIgnoredLines(0),
+            );
+
+        $handler = new ImportTableFromTableHandler($this->clientManager);
+        $handler->setInternalLogger($this->log);
+        /** @var TableImportResponse $response */
+        $response = $handler(
+            $this->projectCredentials,
+            $cmd,
+            [],
+            new RuntimeOptions(['runId' => $this->testRunId]),
+        );
+
+        $this->assertSame(3, $response->getImportedRowsCount());
+        $this->assertSame(
+            ['col1', 'col2'],
+            iterator_to_array($response->getImportedColumns()),
+        );
+        $destinationRef = new BigqueryTableReflection($bqClient, $bucketDatabaseName, $destinationTableName);
+        $this->assertSame(3, $destinationRef->getRowsCount());
+
+        $bqClient->runQuery($bqClient->query(
```

Comment:
imho by se ten cleanup mel stat na zacatku (s trycatchem, ze to nic nesmaze), protoze pokud cestou neco spadne, tak pri dalsim behu tam bude blocker